compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1163,6 +1163,426 @@ TR::Register *OMR::X86::TreeEvaluator::cstoreEvaluator(TR::Node *node, TR::CodeG
+      }
+
+   arrLen -= residualLen;
+
+   //Perform CMPSQ, for the rest of the array, which should be a multiple of 8.
+   if (TR::Compiler->target.is64Bit())
+      TR_ASSERT(arrLen % 8 == 0, "64 Bit: arrLen should be a multiple of 8 at this point, the length is %d", arrLen);
+   else
+      TR_ASSERT(arrLen % 4 == 0, "32 Bit: arrLen should be a multiple of 4 at this point, the length is %d", arrLen);
+
+   while (arrLen > 0)
+      {
+         arrLen -= byteLen;
+         if (TR::Compiler->target.is64Bit())
+            generateInstruction(CMPSQ, node, deps, cg);
+         else
+            generateInstruction(CMPSD, node, deps, cg);
+         generateLabelInstruction(JNE4, node, notEqualLabel, cg);
+      }
+
+   //Result 
+   cg->stopUsingRegister(src1AddrRegister);
+   cg->stopUsingRegister(src2AddrRegister);
+   generateRegImmInstruction(MOVRegImm4(), node, resultRegister, 0, cg);
+   generateLabelInstruction(JMP4, node, endResultLabel, cg);
+
+   generateLabelInstruction(LABEL, node, notEqualLabel, cg);
+   generateLabelInstruction(JL4, node, lessThanLabel, cg);
+   generateRegImmInstruction(MOVRegImm4(), node, resultRegister, 1, cg);
+   generateLabelInstruction(JMP4, node, endResultLabel, cg);
+
+   generateLabelInstruction(LABEL, node, lessThanLabel, cg);
+   generateRegImmInstruction(MOVRegImm4(), node, resultRegister, 2, cg);
+   
+   deps->stopAddingConditions();
+
+   generateLabelInstruction(LABEL, node, endResultLabel, deps, cg);
+   endResultLabel->setEndInternalControlFlow();
+
+   node->setRegister(resultRegister);
+
+   cg->decReferenceCount(src1AddrNode);
+   cg->decReferenceCount(src2AddrNode);
+   cg->recursivelyDecReferenceCount(lengthNode);
+
+   return resultRegister;
+   }
+
+static TR::Register *ConstLenArraycmpLenEvaluator(TR::Node *node, TR::CodeGenerator *cg, uintptrj_t arrLen)
+   {
+   TR::Register *resultRegister = cg->allocateRegister(TR_GPR);
+   TR::Node *src1AddrNode = node->getChild(0);
+   TR::Node *src2AddrNode = node->getChild(1);
+   TR::Node *lengthNode = node->getChild(2);
+   
+   int32_t byteLen = 4;
+   if (TR::Compiler->target.is64Bit())
+      byteLen = 8;
+
+   TR::Register *tempSrc1AddrRegister = cg->evaluate(src1AddrNode);;
+   TR::Register *tempSrc2AddrRegister = cg->evaluate(src2AddrNode);
+   TR::Register *tempArrayLenRegister = cg->evaluate(lengthNode);
+
+   TR::Register *src1AddrRegister = cg->allocateRegister();
+   TR::Register *src2AddrRegister = cg->allocateRegister();
+   TR::Register *arrayLenRegister = cg->allocateRegister();
+
+   TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *residualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *notEqualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *backtrackLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *equalLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *endResultLabel = generateLabelSymbol(cg);
+
+   //Move into registers to for use
```

Comment:
These moves are done regardless of the lifetime of the original register (i.e., if the ref count is one you don't need to copy it).  I think a `gprClobberEvaluate` instead of an `evaluate` on these nodes is what you want, as you'll get the copy done for you.